### PR TITLE
Add support for ember gjs/gts file format

### PR DESCRIPTION
**Pull request by vstefanovic97**
This is a feature request to add support for Ember.js new file format

For more detailed context please read https://community.sonarsource.com/t/add-support-for-embers-new-gjs-gts-file-format-to-sonarjs/103769

To support the new gjs/gts files for linting we use the [ember-eslint-parser](https://github.com/NullVoxPopuli/ember-eslint-parser) used by the official [eslint-plugin-ember](https://github.com/ember-cli/eslint-plugin-ember) library.

I tried to add tests and follow code style as much as possible and all tests seem to be passing on my local.
I'm open to any feedback on what it might take to get this merged and will try to provide as much help possible to extend sonarjs framework support to also include new Ember.js syntax.

Even though Ember.js is not as popular as React, Vue etc. It still has a solid community and ppl using it commercially especially on big projects and I think it makes sense to try and continue to support it

### CODE REVIEW

**Comment by vstefanovic97** [comment]
Hi it's been already 2 weeks since the PR was created can I get any sort of feedback on this?

**Comment by yassin-kammoun-sonarsource** [comment]
Hey @vstefanovic97,

Sorry for the delay, I answered your feature request on Sonar community.

Cheers,

File: package.json [hunk 1]
```diff
@@ -80,6 +80,7 @@
     "@typescript-eslint/parser": "6.14.0",
     "builtin-modules": "3.3.0",
     "bytes": "3.1.2",
+    "ember-eslint-parser": "0.2.5",
```
Comment (NullVoxPopuli) [inline package.json:83]: Can this issue latest? And a caret range? There are a good number of bugfixes that have landed

**Comment by NullVoxPopuli** [comment]
@yassin-kammoun-sonarsource i understand that supporting a whole framework is a big endeavor, but this isn't that! ❤️ 

this is adding support for a file syntax only, for which all editors, highlighters, and even github already support.

If you have any questions or concerns, please don't hesitate to ask

**Comment by yassin-kammoun-sonarsource** [comment]
@NullVoxPopuli I will discuss again the topic with my teammates, but I can't promise anything :)

**Comment by SergeAstapov** [comment]
@yassin-kammoun-sonarsource thank you! Reviewing changes in this PR again, tbh there does not seem to be much (if any) framework specific code beside adding .gjs/.gts file extensions and wiring up `eslint-plugin-ember`. Half of the changes are adding tests  and respective deps.

All the heavy lifting is done in respective ESLint plugins and `eslint-plugin-ember` does just that.

Please consider this as an argument for discussion 👍

**Comment by yassin-kammoun-sonarsource** [comment]
Hey there,

We discussed the topic with the team and the conclusion is the same: we are not going to invest in supporting Ember.js for now.

For more information, please refer to the [latest update](https://community.sonarsource.com/t/add-support-for-embers-new-gjs-gts-file-format-to-sonarjs/103769/14?u=yassin) from my colleague.

Cheers!

**Comment by runspired** [comment]
@yassin-kammoun-sonarsource this is effectively a 40 line change, being made by a community to an ostensibly open source project to support itself.

There is no investment being made on your part, the maintenance on this would be laughably low if it ever occurred. 

On the other hand, the damage to sonarcloud could be high. EmberJS represents a large number of top-tier SAAS and financial products, and we will be forced to leave.

**Pull request closed without merging**